extends Control

#icon
#name
#unit type
#popularity dic
#nutrient dic
var nutrient = preload("res://nutrient_item.tscn")
#var emojis = "ðŸðŸŽðŸðŸŠðŸ‹ðŸ‹â€ðŸŸ©ðŸŒðŸ‰ðŸ‡ðŸ“ðŸ«ðŸˆðŸ’ðŸ‘ðŸ¥­ðŸðŸ¥¥ðŸ¥ðŸ…ðŸ†ðŸ¥‘ðŸ«›ðŸ¥¦ðŸ¥¬ðŸ¥’ðŸŒ¶ðŸ«‘ðŸŒ½ðŸ¥•ðŸ«’ðŸ§„ðŸ§…ðŸ¥”ðŸ ðŸ«šðŸ¥ðŸ¥¯ðŸžðŸ¥–ðŸ¥¨ðŸ§€ðŸ¥šðŸ³ðŸ§ˆðŸ¥žðŸ§‡ðŸ¥“ðŸ¥©ðŸ—ðŸ–ðŸ¦´ðŸŒ­ðŸ”ðŸŸðŸ•ðŸ«“ðŸ¥ªðŸ¥™ðŸ§†ðŸŒ®ðŸŒ¯ðŸ«”ðŸ¥—ðŸ¥˜ðŸ«•ðŸ¥«ðŸ«™ðŸðŸœðŸ²ðŸ›ðŸ£ðŸ±ðŸ¥ŸðŸ¦ªðŸ¤ðŸ™ðŸšðŸ˜ðŸ¥ðŸ¥ ðŸ¥®ðŸ¢ðŸ¡ðŸ§ðŸ¨ðŸ¦ðŸ¥§ðŸ§ðŸ°ðŸŽ‚ðŸ®ðŸ­ðŸ¬ðŸ«ðŸ¿ðŸ©ðŸªðŸŒ°ðŸ¥œðŸ«˜ðŸ¯ðŸ¥›ðŸ«—ðŸ¼ðŸ«–â˜•ï¸ðŸµðŸ§ƒðŸ¥¤ðŸ§‹ðŸ¶ðŸºðŸ»ðŸ¥‚ðŸ·ðŸ¥ƒðŸ¸ðŸ¹ðŸ§‰ðŸ¾ðŸ§ŠðŸ¥„ðŸ´ðŸ½ðŸ¥£ðŸ¥¡ðŸ¥¢ðŸ§‚ðŸŒ¾ðŸª»ðŸ¦‘ðŸ€ðŸŒ¿ðŸŒ±ðŸ›ŸðŸŽâšœï¸ðŸ©·ðŸðŸ“ðŸ’ðŸŽðŸ‰ðŸ‘ðŸŠðŸ¥­ðŸðŸŒðŸ‡ðŸ«ðŸ«’ðŸ¥ðŸðŸðŸˆðŸ‹â€ðŸŸ©ðŸ‹ðŸ¥¥ðŸ…ðŸŒ¶ï¸ðŸ«šðŸ¥•ðŸ§…ðŸŒ½ðŸ¥¦ðŸ¥’ðŸ„â€ðŸŸ«ðŸ¥”ðŸ§„ðŸ†ðŸ ðŸ¥‘ðŸ«‘ðŸ«›ðŸ¥¬ðŸ«˜ðŸŒ°ðŸ¥œðŸžðŸ«“ðŸ¥ðŸ¥–ðŸ¥¯ðŸ§‡ðŸ”ðŸ–ðŸ—ðŸ¥©ðŸ¥“ðŸ§€ðŸ¥šðŸ³ðŸ¥žðŸŒ­ðŸ¥ªðŸ¥¨ðŸŸðŸ•ðŸ«”ðŸŒ®ðŸŒ¯ðŸ¥™ðŸ›ðŸ²ðŸ¥—ðŸ¥£ðŸ«•ðŸ¥«ðŸðŸ¥˜ðŸ§†ðŸœðŸ¦ªðŸ¦žðŸ£ðŸ¤ðŸ¥¡ðŸšðŸ±ðŸ¥ŸðŸ¢ðŸ™ðŸ˜ðŸ¥ðŸ¡ðŸ¥ ðŸ¥®ðŸ§ðŸ¨ðŸ«ðŸ¬ðŸ­ðŸ§ðŸŽ‚ðŸ®ðŸ°ðŸ¥§ðŸ¦ðŸ©ðŸªðŸ¯ðŸ§‚ðŸ§ˆðŸ¿ðŸ§ŠðŸ«™ðŸ¥¤ðŸºðŸ§‰ðŸ«–â˜•ï¸ðŸµðŸ¼ðŸ¥›ðŸ§ƒðŸ§‹ðŸ»ðŸ¥‚ðŸ¾ðŸ·ðŸ¥ƒðŸ«—ðŸ¸ðŸ¹ðŸ¶ðŸ½ï¸ðŸ”ªðŸ¥„ðŸ´ðŸ¥¢"
var emojiList = ["ðŸ", "ðŸŽ", "ðŸ", "ðŸŠ", "ðŸ‹", "â€", "ðŸŸ©", "ðŸŒ", "ðŸ‰", "ðŸ‡", "ðŸ“", "ðŸ«", "ðŸˆ", "ðŸ’", "ðŸ‘", "ðŸ¥­", "ðŸ", "ðŸ¥¥", "ðŸ¥", "ðŸ…", "ðŸ†", "ðŸ¥‘", "ðŸ«›", "ðŸ¥¦", "ðŸ¥¬", "ðŸ¥’", "ðŸŒ¶", "ðŸ«‘", "ðŸŒ½", "ðŸ¥•", "ðŸ«’", "ðŸ§„", "ðŸ§…", "ðŸ¥”", "ðŸ ", "ðŸ«š", "ðŸ¥", "ðŸ¥¯", "ðŸž", "ðŸ¥–", "ðŸ¥¨", "ðŸ§€", "ðŸ¥š", "ðŸ³", "ðŸ§ˆ", "ðŸ¥ž", "ðŸ§‡", "ðŸ¥“", "ðŸ¥©", "ðŸ—", "ðŸ–", "ðŸ¦´", "ðŸŒ­", "ðŸ”", "ðŸŸ", "ðŸ•", "ðŸ«“", "ðŸ¥ª", "ðŸ¥™", "ðŸ§†", "ðŸŒ®", "ðŸŒ¯", "ðŸ«”", "ðŸ¥—", "ðŸ¥˜", "ðŸ«•", "ðŸ¥«", "ðŸ«™", "ðŸ", "ðŸœ", "ðŸ²", "ðŸ›", "ðŸ£", "ðŸ±", "ðŸ¥Ÿ", "ðŸ¦ª", "ðŸ¤", "ðŸ™", "ðŸš", "ðŸ˜", "ðŸ¥", "ðŸ¥ ", "ðŸ¥®", "ðŸ¢", "ðŸ¡", "ðŸ§", "ðŸ¨", "ðŸ¦", "ðŸ¥§", "ðŸ§", "ðŸ°", "ðŸŽ‚", "ðŸ®", "ðŸ­", "ðŸ¬", "ðŸ«", "ðŸ¿", "ðŸ©", "ðŸª", "ðŸŒ°", "ðŸ¥œ", "ðŸ«˜", "ðŸ¯", "ðŸ¥›", "ðŸ«—", "ðŸ¼", "ðŸ«–", "â˜•", "ï¸", "ðŸµ", "ðŸ§ƒ", "ðŸ¥¤", "ðŸ§‹", "ðŸ¶", "ðŸº", "ðŸ»", "ðŸ¥‚", "ðŸ·", "ðŸ¥ƒ", "ðŸ¸", "ðŸ¹", "ðŸ§‰", "ðŸ¾", "ðŸ§Š", "ðŸ¥„", "ðŸ´", "ðŸ½", "ðŸ¥£", "ðŸ¥¡", "ðŸ¥¢", "ðŸ§‚", "ðŸŒ¾", "ðŸª»", "ðŸ¦‘", "ðŸ€", "ðŸŒ¿", "ðŸŒ±", "ðŸ›Ÿ", "ðŸŽ", "âšœ", "ðŸ©·", "ðŸ„", "ðŸŸ«", "ðŸ¦ž", "ðŸ”ª"]

# Called when the node enters the scene tree for the first time.
func _ready():
	#for i in emojis:
		#if not emojiList.has(i):
			#emojiList.append(i)
	#print(emojiList)
	for i in emojiList:
		var emojiButton = Button.new()
		emojiButton.text = i
		emojiButton.pressed.connect(emojiSelect.bind(emojiButton.text))
		#emojiButton.add_theme_font_size_override('emojiButton', 75)
		#emojiButton.size = Vector2(300,300)
		emojiButton.add_theme_font_size_override("font_size", 35)
		$PanelContainer/MarginContainer/emojiGrid.add_child(emojiButton)
	$PanelContainer.position.y = 2000
	
	
	#http_data(2118224) #cheese or something
	#http_data(2626748) #BB power waffle
	#http_data(2489244)

func _notification(what):
	if what == 1007:
		if $PanelContainer.visible:
			$AnimationPlayer.play('slideDown')
		else:
			get_tree().change_scene_to_file("res://main.tscn")

func emojiSelect(text):
	$VBoxContainer/HBoxContainer/emojiButton.text = text
	$AnimationPlayer.play('slideDown')

# Called every frame. 'delta' is the elapsed time since the previous frame.
func _process(delta):
	pass


func _on_button_pressed():
	var nut = nutrient.instantiate()
	$VBoxContainer/PanelContainer/ScrollContainer/MarginContainer/VBoxContainer/VBoxContainer.add_child(nut)


func _on_sub_pressed():
	if $VBoxContainer/PanelContainer/ScrollContainer/MarginContainer/VBoxContainer/VBoxContainer.get_child_count() > 0:
		$VBoxContainer/PanelContainer/ScrollContainer/MarginContainer/VBoxContainer/VBoxContainer.get_child(-1).queue_free()


func _on_emoji_button_pressed():
	$AnimationPlayer.play('slideUp')
	print('hard pressed for answers?')


func _on_popup_menu_id_pressed(id):
	$USDAcontainer/VBoxContainer/HBoxContainer/VBoxContainer/databaseType.text = $USDAcontainer/VBoxContainer/HBoxContainer/VBoxContainer/databaseType/PopupMenu.get_item_text(id)


func _on_database_type_pressed():
	$USDAcontainer/VBoxContainer/HBoxContainer/VBoxContainer/databaseType.visible = true

func http_data(input):
	$HTTPRequest.request_completed.connect(_on_request_completed)
	$HTTPRequest.request("https://api.nal.usda.gov/fdc/v1/food/" + str(input) +"?api_key=DEMO_KEY")
	#$HTTPRequest.request("https://api.nal.usda.gov/fdc/v1/foods/search?api_key=DEMO_KEY&query=thick%20and%20fluffy%20power%20waffle&pageSize=100&dataType=Branded&brandOwner=Baker%20Mills")
	#$HTTPRequest.request("https://api.nal.usda.gov/fdc/v1/foods/search?api_key=DEMO_KEY&query=HIGH%20PROTEIN%20SHAKE%20CHOCOLATE&pageSize=100&dataType=Branded")

func http_search():
	$HTTPRequest.request_completed.connect(_on_request_completed)
	$HTTPRequest.request("https://api.nal.usda.gov/fdc/v1/foods/search?api_key=DEMO_KEY&query=Cheddar%20Cheese")

func _on_request_completed(result, response_code, headers, body):
	print(response_code)
	var json = JSON.parse_string(body.get_string_from_utf8())
	print(json)
	#print(len(json['foods']))
	#for i in json['foods']:
		#print(i["description"])
		
	#global.currentDeck = json["values"]
	#global.currentDims = global.currentDeck.pop_front()
	#global.currentDeck.shuffle()
	
	#data comes in as list of lists: json["values"][0][0] gives first row first item
	#get_tree().change_scene_to_file("res://card_viewer.tscn")
	
	


func _on_food_name_focus_entered():
	$USDAcontainer/VBoxContainer/searchTips.visible = true


func _on_food_name_focus_exited():
	$USDAcontainer/VBoxContainer/searchTips.visible = false
